hypervisor/src/mshv/mod.rs

```diff
@@ -568,8 +568,34 @@ impl cpu::Vcpu for MshvVcpu {
     ///
     /// X86 specific call to setup the CPUID registers.
     ///
-    fn set_cpuid2(&self, _cpuid: &[CpuIdEntry]) -> cpu::Result<()> {
-        Ok(())
+    fn set_cpuid2(&self, cpuid: &[CpuIdEntry]) -> cpu::Result<()> {
+        let mut ret = Ok(());
+
+        for entry in cpuid.iter() {
+            let ent = hv_cpuid_entry::from(*entry);
+
+            let override_arg = None;
+            let mut idx = None;
+            let mut subleaf_specific = None;
+
+            match entry.function {
```

Comment:
Regarding the filtering - it's needed to pass the arguments specific to these particular leaves. The hypercall arguments are documented here:

https://github.com/rust-vmm/mshv/blob/main/mshv-ioctls/src/ioctls/vcpu.rs#L941

For the semantic - yes, it could be made closer to KVM by integrating [register_intercept_result_cpuid](https://github.com/rust-vmm/mshv/blob/main/mshv-ioctls/src/ioctls/vcpu.rs#L986), however all the extra conditions would have to go into there (currently just these two cases).

With both points, why I chose this approach is because further changes that might need some deviating arguments would have to be split between the `mshv` crate and `cloud-hypervisor`. With the current approach, this kind of changes would land here, too. If we want to pursue an approach to encapsulate more on the mshv crate side instead of here, I'll have to reorganize this PR by splitting this part accordingly. Please let me know.

Thanks

Reply:
After digging more, it seems that moving the mentioned parts into the `mshv` crate is the right solution. I'm going to move this PR to draft and split off the filtering part into the `mshv` crate, than continue on this side. The `mshv` crate is the piece that theoretically might be consumed else where, so seems that's the right place that should know what to do with the special additional options.

Thanks

Reply:
I filed a [PR to mshv crate](https://github.com/rust-vmm/mshv/pull/87) to move this piece functionality. Will continue here once it's complete.

Thanks